File: BLEVersion/JavaVersion/Scan/Presenter/ScanPresenter.java
package com.gravicode.neospectralib.Scan.Presenter;

import com.gravicode.neospectralib.BluetoothSDK.SWS_P3Packet;

import static com.gravicode.neospectralib.Global.GlobalVariables.bluetoothAPI;

/**
 *
 */

public class ScanPresenter /*implements IScanPresenter */ {

    private String TAG = getClass().getSimpleName();


    public void requestSensorReading(int scanTime) {
        bluetoothAPI.setSourceSettings();
        bluetoothAPI.setOpticalSettings();
        bluetoothAPI.sendDefaultPacket(scanTime);
    }

    public void requestBackgroundReading(int scanTime) {
        bluetoothAPI.setSourceSettings();
        bluetoothAPI.setOpticalSettings();
        bluetoothAPI.sendBackgroundPacket(scanTime);
    }



    public void requestSelfCalibration(int scanTime) {
        bluetoothAPI.setSourceSettings();
        bluetoothAPI.setOpticalSettings();
        bluetoothAPI.sendSelfCalibrationPacket(scanTime);
    }

    public void restoreToDefaultSettings()
    {
        bluetoothAPI.sendRestoreToDefaultPacket();
    }

    public void storingSettings()
    {
        bluetoothAPI.sendStoreGainPacket();
        bluetoothAPI.sendStoreSelfCorrectionPacket();

    }

    public void requestGainReading() {
        bluetoothAPI.sendGainAdjustmentPacket();
    }

    public void sendCustomPacket(SWS_P3Packet newPacket, String BLE_Service) {
        bluetoothAPI.sendPacket(newPacket.getPacketStream(), newPacket.isInterpolationMode(), BLE_Service);

    }

    public void sendOTACommand(SWS_P3Packet newPacket) {
        bluetoothAPI.sendOTAPacket(newPacket.getPacketStream());
    }


}

File: BLEVersion/DotnetVersion/NeospectraApp/NeospectraApp/Driver/Scan/ScanPresenter.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Runtime.CompilerServices;
using System.Text;
using System.Threading.Tasks;

namespace NeospectraApp.Driver
{
    public class ScanPresenter /*implements IScanPresenter */
    {
        
        private String TAG = nameof(ScanPresenter);

        public ScanPresenter()
        {
            Setup();
        }

        async void Setup()
        {
            var res = await GlobalVariables.bluetoothAPI.setNotifications();
        }
        public void requestSensorReading(int scanTime)
        {

            GlobalVariables.bluetoothAPI?.setSourceSettings();
            GlobalVariables.bluetoothAPI?.setOpticalSettings();
            GlobalVariables.bluetoothAPI?.sendDefaultPacket(scanTime);
        }

        public void requestBackgroundReading(int scanTime)
        {
            GlobalVariables.bluetoothAPI?.setSourceSettings();
            GlobalVariables.bluetoothAPI?.setOpticalSettings();
            GlobalVariables.bluetoothAPI?.sendBackgroundPacket(scanTime);
        }



        public void requestSelfCalibration(int scanTime)
        {
            GlobalVariables.bluetoothAPI?.setSourceSettings();
            GlobalVariables.bluetoothAPI?.setOpticalSettings();
            GlobalVariables.bluetoothAPI?.sendSelfCalibrationPacket(scanTime);
        }

        public void restoreToDefaultSettings()
        {
            GlobalVariables.bluetoothAPI?.sendRestoreToDefaultPacket();
        }

        public void storingSettings()
        {
            GlobalVariables.bluetoothAPI?.sendStoreGainPacket();
            GlobalVariables.bluetoothAPI?.sendStoreSelfCorrectionPacket();

        }

        public void requestGainReading()
        {
            GlobalVariables.bluetoothAPI?.sendGainAdjustmentPacket();
        }

        public void sendCustomPacket(SWS_P3Packet newPacket, String BLE_Service)
        {
            GlobalVariables.bluetoothAPI?.sendPacket(newPacket.getPacketStream(), newPacket.isInterpolationMode(), BLE_Service);

        }

        public void sendOTACommand(SWS_P3Packet newPacket)
        {
            GlobalVariables.bluetoothAPI?.sendOTAPacket(newPacket.getPacketStream());
        }


    }

}
